src/Build/Evaluation/Evaluator.cs

```diff
@@ -243,6 +244,11 @@ internal class Evaluator<P, I, M, D>
 #endif
 
         /// <summary>
+        /// The search paths are machine specific and should not change during builds
+        /// </summary>
+        private static readonly EngineFileUtilities.IOCache _fallbackSearchPathsCache = new EngineFileUtilities.IOCache();
```

Comment:
We discussed this today. The directories that are cached here are: the VS instance's MSBuild directory, and the program files MSBuild directory (just those directories, not their subdirectories). Both should require VS being closed to be created, so the process-lifetime cache should be fine.